src/main/java/uk/gov/pay/connector/tasks/HistoricalEventEmitterByDateRangeTask.java

```
@@ -43,28 +49,31 @@ public HistoricalEventEmitterByDateRangeTask(HistoricalEventEmitterWorker worker
+        try {
+            Optional<ZonedDateTime> startDate = getDateParam(parameters, "start_date");
+            Optional<ZonedDateTime> endDate = getDateParam(parameters, "end_date");
+            final Long doNotRetryEmitUntilDuration = getDoNotRetryEmitUntilDuration(parameters);
 
-        logger.info("Execute called startDate={} endDate={} - processing", startDate, endDate);
+            logger.info("Execute called startDate={} endDate={}, doNotRetryEmitUntilDuration={} - processing",
+                    startDate, endDate, doNotRetryEmitUntilDuration);
 
+            if (startDate.isEmpty() || endDate.isEmpty()) {
+                logger.info("Rejected request, both start date and end date are mandatory");
+                output.println("Rejected request, both start date and end date are mandatory");
+            } else {
 
-        if (startDate.isEmpty() || endDate.isEmpty()) {
-            logger.info("Rejected request, both start date and end date are mandatory");
-            output.println("Rejected request, both start date and end date are mandatory");
-        } else {
-            try {
                 logger.info("Request accepted");
-                executor.execute(() -> worker.executeForDateRange(startDate.get(), endDate.get()));
+                executor.execute(() -> worker.executeForDateRange(startDate.get(), endDate.get(),
+                        doNotRetryEmitUntilDuration));
                 output.println("Accepted");
-            } catch (java.util.concurrent.RejectedExecutionException e) {
-                logger.info("Rejected request, worker already running");
-                output.println("Rejected request, worker already running");
             }
+        } catch (java.util.concurrent.RejectedExecutionException | NumberFormatException e) {
+            logger.info("Rejected request, worker already running");
+            output.println("Rejected request, worker already running");
```

[Question] Why was the `try/catch` moved?

---

It is to handle invalid `do_not_retry_emit_until` and return `Rejected` response instead of returning 500

---

Yeah...I guess it't just the error message is not clear in case of the parameter being invalid.
Probably could be addressed as part of refactoring parameter extraction.

---

There is lots around the tasks to improve for validation / refactoring. Would prefer a separate story to improve it all